clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy

```diff
@@ -50,6 +51,13 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
 
     def autoscalerName = description.serverGroupName
     def parsedName = Names.parseName(description.serverGroupName)
+
+    if (description.kind) {
+      if (description.kind == KubernetesUtil.CONTROLLERS_STATEFULSET_KIND || description.kind == KubernetesUtil.CONTROLLERS_DAEMONSET_KIND) {
```

Comment:
You seem to be checking these quite a bit - ideally to keep the code easy to reason about I would recommend having one switch statement per-operation that decides how to carry out the delete, rather than `if` statements tucked away in several nested functions.

Reply:
sure. I will create method to do the checking.